ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -2280,6 +2281,16 @@ static vk_device ggml_vk_get_device(size_t idx) {
             device->max_memory_allocation_size = props3.maxMemoryAllocationSize;
         }
 
+        const char* GGML_VK_SUBALLOCATION_BLOCK_SIZE = getenv("GGML_VK_SUBALLOCATION_BLOCK_SIZE");
+
+        if (GGML_VK_SUBALLOCATION_BLOCK_SIZE != nullptr) {
+            device->suballocation_block_size = std::stoul(GGML_VK_SUBALLOCATION_BLOCK_SIZE);
+        } else {
+            // Limit batching of allocations to 1GB by default to avoid fragmentation issues
+            device->suballocation_block_size = 1024*1024*1024;
```

Comment:
I'd still rather not default to this, since it can reduce performance. I have not run into this issue on other platforms yet (but I admit most of my testing happens on GPUs that don't run a desktop at the same time). Maybe set defaults by vendor and OS and we tune this as required?

Reply:
Why do you think it will reduce performance? Even with very large page sizes and TLB compression, 1GB is quite large. VMA for example uses a 256MB block size by default (https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/blob/master/include/vk_mem_alloc.h#L3289).

Reply:
I remember reports about this in the past, when we had to deal with wrong max allocation sizes reported on AMD Windows, so it stuck in my head to allocate chunks as big as possible. Maybe that's not true.

But I can also see a small negative performance difference with Intel on this branch.

Reply:
ok, I've updated the change to only apply to NVIDIA drivers on Windows.